solutions/usaco-988.mdx

```diff
@@ -7,12 +7,71 @@ author: Ananth Kashyap
 
 [Official Analysis (C++)](http://www.usaco.org/current/data/sol_photo_bronze_jan20.html)
 
+Since $N \leq 1000$, we can simulate every ordering and pick the first valid permutation (since that would be the lexicographical minimum).
 ## Implementation
 
-**Time Complexity:** $\mathcal{O}(N^3)$
+**Time Complexity:** $\mathcal{O}(N^2)$
 
 <LanguageSection>
 
+<CPPSection>
+
+```cpp 
+#include <iostream>
+#include <vector>
+#include <algorithm>
+
+using namespace std;
+
+int main() {
+	freopen("photo.in", "r", stdin);
+	freopen("photo.out", "w", stdout);
+
+	int n; cin >> n;
+	vector<int> order;
+
+	for (int i = 1; i < n; i++) {
+		int a; cin >> a;
+		order.push_back(a);
+	}
+	
+	// count represents the first element.
+	int count = 1;
+	vector<int> ans{count};
+	while (true) {
+		for (int i = 0; i < order.size(); i++) {
+			// append the next element that follows the order.
+			ans.push_back(order.at(i) - ans.at(ans.size() - 1));
+		}
+
+		bool flag = false;
+		for (int i = 0; i < ans.size(); i++) {
+			if (ans.at(i) <= 0 || ans.at(i) > n) {
+				flag = true;
+				break;
+			}
+		}
+
+		if (!unique(ans.begin(), ans.end() == ans.end()) || flag) {
```

Comment:
```suggestion
		if (unique(ans.begin(), ans.end()) != ans.end() || flag) {
```
shouldn't it be like this?

Reply:
they both work, but that's cleaner :)

Reply:
i think your initial one was right but the one sans changed it to is incorrect

Reply:
yeah i was bad lmao